File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    # get height from user
    while True:
        height = get_int("Height: ")

        if height > 0 and height <= 8:
            break

    # print pyramid
    for i in range(1, height + 1):
        spaceCount = height - i

        # print spaces, then left pyramid, then 2-space gap, then right pyramid
        print(" " * spaceCount, "#" * i, "  ", "#" * i, sep="")


if __name__ == "__main__":
    main()

File: pset1/mario/less/mario.c
// pset1: Mario (less)
// create a program, utilising printf(), that prints out a half-pyramid of a user-defined height, displayed as #
// e.g.
//     #
//    ##
//   ###
//  ####
// #####

// Specifications: integer between 1 and 8

#include <cs50.h> // for get_int()
#include <stdio.h>

int main(void)
{
    // do-while loop
    int h;
    do
    {
        // prompt once for integer h
        h = get_int("Height: ");
    }
    // reprompt if h is not between 1 and 8
    while (h < 1 || h > 8);

    // define y, which will be used, as a separate decreasing counter with an initial value h, to determine number of spaces
    int y = h;

    // print pyramid using for loop
    // define number of lines (pyramid height) based on h
    for (int i = 0; i < h; i++)
    {
        // print spaces
        for (int a = 1; a < y; a++)
        {
            printf(" ");
        }
        //after exiting loop (or effectively, one pyramid row), decrease value of y by 1
        y--;

        // print hashtags
        for (int b = 0; b < i + 1; b++)
        {
            printf("#");
        }

        // next line
        printf("\n");
    }

}